client/client.go

```diff
@@ -177,56 +166,43 @@ func (c *LogClient) waitForRootUpdate(ctx context.Context) error {
 // UpdateRoot retrieves the current SignedLogRoot.
 // Verifies the signature, and the consistency proof if this is not the first root this client has seen.
 func (c *LogClient) UpdateRoot(ctx context.Context) error {
-	req := &trillian.GetLatestSignedLogRootRequest{
-		LogId: c.LogID,
-	}
-	resp, err := c.client.GetLatestSignedLogRoot(ctx, req)
+	resp, err := c.client.GetLatestSignedLogRoot(ctx,
+		&trillian.GetLatestSignedLogRootRequest{
+			LogId: c.LogID,
+		})
 	if err != nil {
 		return err
 	}
-	str := resp.SignedLogRoot
-
-	// Verify SignedLogRoot signature.
-	hash := crypto.HashLogRoot(*str)
-	if err := crypto.Verify(c.pubKey, hash, str.Signature); err != nil {
-		return err
-	}
-
-	// Verify Consistency proof.
-	if str.TreeSize == c.root.TreeSize &&
-		bytes.Equal(str.RootHash, c.root.RootHash) {
+	if resp.SignedLogRoot.TreeSize == c.root.TreeSize &&
+		bytes.Equal(resp.SignedLogRoot.RootHash, c.root.RootHash) {
```

Comment:
This if statement is just a short circuit when nothing has changed. 
eg. Running UpdateRoot, which returns the same treesize and root hash. 

Hashes of equally sized roots should match, but if they don't, we want to be sure we return an error. "Trust but verify" :-)